Sources/Utilities/Storage/Session/SessionManager.swift

```diff
@@ -21,7 +21,7 @@ protocol SessionManager: UserProvider {
     var localAuthentication: LocalAuthenticationManager { get }
 
     /// Starts a new session from a remote login
-    func startSession(using session: LoginSession) async throws
+    func startSession(using session: LoginSession, configurationInitialiser: LoginSessionConfigurationProvider.Type) async throws
```

Comment:
We need to pass the `persistentID` into the `LoginSessionConfiguration` though, I suppose we could make it as part of the protocol `LoginSession` and create the `LoginSessionConfiguration` in the `AuthenticationCoordinator` but yeah.
This felt cleaner because we can pass a mock to create whatever configuration we want

Reply:
Hm 🤔 I do think it makes more sense to inject in from the coordinator because we have the flexibility without needing the complexity of different types. It feels a bit overly complex to need a protocol just to create an instance of a configuration, but not a hill I am willing to die on 🙂 whatever makes sense to you & Magdaline!

Reply:
I totally agree, it would be way nicer if we could just pass the instance of `LoginSessionConfiguration` to this `PersistentSessionManager.startSession()` method.
Though, an instance of `LoginSessionConfiguration` needs to be initialised with a `persistentSessionID`, that property only exists on the `PersistentSessionManager` (the class we're passing this instance of `LoginSessionConfiguration` into) so unless we go and make that property a variable in the authentication package and set it in on the instance within the body of `startSession()` then I don't know how we would be able to get the `persistentSessionID` into the `LoginSessionConfiguration` used for the login session?
The way it's currently set up where we pass in a type which can initialise a configuration felt good because we can mock the methods to provide a mock configuration instance. Totally happy to change it if we think there's a better way but I don't think we can pass in an instance of `LoginSessionConfiguration` in it's current form?